doc(pe): Document policy options (#601)
armory/docs: content/en/docs/plugin-guide/plugin-policy-engine/packages/spinnaker.http.authz/tasks/_index.md

---
title: "Tasks"
linkTitle: "Tasks"

> description: "Posts to the tasks api create new tasks in Spinnaker."

Reviewer: ```suggestion
description: "Posts to the tasks API create new tasks in Armory Enterprise."
```

---

- **Path:** tasks
- **Method:** Post
- **Package:** `spinnaker.http.authz`

The following rego function can be leveraged in any task to determine what task type is being created:

{{< prism lang="rego" line-numbers="true" >}}
    createsTaskOfType(tasktype){
        input.method="POST"
        input.path=["tasks"]
        input.body.job[_].type=tasktype
    }
{{< /prism >}}

## Keys

| Key            |   Type   | Description                                                                                                                                                                                                                                                                                                    |
| :------------- | :------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `input.body`   | `object` | The contents of this field will depend on the task type. It corresponds to the payload body being posted to the spinnaker API for that task. If your policy needs to conditionally enable/disable a task based off a property being configured by that task, the value you need will likely be in this object. |

Reviewer: ```suggestion
| `input.body`   | `object` | The contents of this field depends on the task type. It corresponds to the payload body being posted to the API for that task. If your policy needs to conditionally enable/disable a task based off a property being configured by that task, the value you need is likely in this object. |
```

> | `input.method` | `string` | The HTTP method by which the API is being called. When createing a task this will be 'POST'.                                                                                                                                                                                                                   |

Reviewer: ```suggestion
| `input.method` | `string` | The HTTP method by which the API is being called. When creating a task, this will be 'POST'.                                                                                                                                                                                                                   |
```
